### ISSUE TITLE
Command line mode mapping to <down> not working with wildmenu.

### ISSUE THREAD

**Issue by dstein64** [vim/vim#6045]
**Describe the bug**

The documentation for `wildmenumode()` provides an example to make `<c-j>` work like `<down>` in wildmode:
```
:cnoremap <expr> <C-j> wildmenumode() ? "\<Down>\<Tab>" : "\<c-j>"
```

After creating that mapping, using `<c-j>` when the wildmenu is shown results in `^I` being appended to the command line command.

The following mapping—intended to highlight the issue as opposed to be useful on its own—also encounters the problem.
```
:cnoremap j <Down>
```

**To Reproduce**

Detailed steps to reproduce the behavior:
1. Run `vim --clean`
2. Create mapping: `:cnoremap <expr> <C-j> wildmenumode() ? "\<Down>\<Tab>" : "\<c-j>"`
3. Type `:e /` followed by `<tab>` to bring up the wildmenu for the root directory.
4. Press `<C-j>`, which should "work like `<down>`".
5. `^I` is appended to the command (twice), as opposed to having the same behavior as pressing `<down>`.

**Expected behavior**

I was expecting a move into the selected subdirectory, which is what happens when pressing `<down>`.

**Environment (please complete the following information):**

 - Vim version: `8.2.700`
 - OS: Ubuntu 19.10
 - Terminal: GNOME Terminal 3.34.2

**Comment by lacygoill** [comment]
Possible duplicate of: https://github.com/vim/vim/issues/4954

For a workaround, make sure that `'wildcharm'` has the same value as `'wildchar'` (whose default is `<Tab>`/`9`):

    vim -Nu NONE +'set wildmenu wildcharm=9' +'cnoremap <expr> <C-j> wildmenumode() ? "\<Down>\<Tab>" : "\<c-j>"'
                                ^^^^^^^^^^^

**Comment by lacygoill** [comment]
Or use `feedkeys()` with the `t` flag:

    vim -Nu NONE +'set wildmenu' +'cnoremap <expr> <C-j> wildmenumode() ? feedkeys("\<down>\<tab>", "t")[-1] : "\<c-j>"'
                                                                                                     ^
See also:

- https://groups.google.com/d/msg/vim_dev/xf5TRb4uR4Y/djk2dq2poaQJ
- https://stackoverflow.com/q/14842987/9780968

**Comment by dstein64** [comment]
Thanks @lacygoill.

**Comment by lacygoill** [comment]
Fixed by [8.2.2221](https://github.com/vim/vim/releases/tag/v8.2.2221).
